app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,107 @@
+class OrdersController < ApplicationController
+
+  def create
+    @product = Product.find(params[:product_id])
+    order
+  
+    begin
+      if @product
+        existing_order_item = @order.order_items.find_by(product: @product)
+      
+        if existing_order_item
+          existing_order_item.quantity += 1
+          if existing_order_item.save
+            redirect_to root_path, notice: 'Product added to cart.'
+          else
+            redirect_to root_path, notice: 'Error updating cart.'
+          end
+        else
+          @order_item = @order.order_items.new(product: @product, quantity: 1, out_of_stock: false)
+          if @order_item.save
+            redirect_to root_path, notice: 'Product added to cart.'
+          else
+            redirect_to root_path, notice: "Error adding product to cart."
+          end
+        end
+      else
+        redirect_to root_path, notice: 'Product not found.'
+      end
+    rescue => ex
```

Comment:
Names should be more representative, so you can use something else rather than `ex`